autorest/azure/cli/token.go

```diff
@@ -112,3 +116,55 @@ func LoadTokens(path string) ([]Token, error) {
 
 	return tokens, nil
 }
+
+// GetTokenFromCli gets a token using Azure CLI 2.0 for local development scenarios.
+func GetTokenFromCli(resource string) (*Token, error) {
+	// This is the path that a develop can set to tell this class what the install path for Azure CLI is.
+	const azureCliPath = "AzureCLIPath"
+
+	// The default install paths are used to find Azure CLI. This is for security, so that any path in the calling program's Path environment is not used to execute Azure CLI.
+	azureCliDefaultPathWindows := fmt.Sprintf("%s\\Microsoft SDKs\\Azure\\CLI2\\wbin; %s\\Microsoft SDKs\\Azure\\CLI2\\wbin", os.Getenv("ProgramFiles(x86)"), os.Getenv("ProgramFiles"))
```

Comment:
Can users install the CLI to a non-default path?  If so then this would be broken.
CC @lmazuel

Reply:
I guess the solution here is the `AzureCLIPath` environment var yes?

Reply:
CLI's doesn't let you select installation path. The logic here is fine. 
Because CLI's installer ensures global PATH env var gets updated for its executable, maybe the fallback would be just run "az" w/o the full path?

Reply:
The idea here is to limit what's on the path so that something doesn't hijack the `az` command.

Reply:
@jhendrixMSFT You're right. `AzureCLIPath` can be set to override the default path.

The .NET Azure SDK team considered PATH hijacking a sufficient risk to not use it.
Simply using PATH _would_ simplify a bit because there would no longer be any os-specific logic.

Should I leave it as is, or simplify the code to simply use the az binary in PATH?

Reply:
I'd leave it as-is to avoid PATH hijacking.

Reply:
Ok! Marking this as resolved then.